rainier-tests/src/test/scala/com/stripe/rainier/core/WAICTest.scala

```diff
@@ -0,0 +1,75 @@
+package com.stripe.rainier.core
+
+import com.stripe.rainier.sampler._
+import org.scalatest.FunSuite
+
+class WAICTest extends FunSuite {
+  implicit val rng = RNG.default
+  val cars = List(
+    (4.0, 2.0),
+    (4.0, 10.0),
+    (7.0, 4.0),
+    (7.0, 22.0),
+    (8.0, 16.0),
+    (9.0, 10.0),
+    (10.0, 18.0),
+    (10.0, 26.0),
+    (10.0, 34.0),
+    (11.0, 17.0),
+    (11.0, 28.0),
+    (12.0, 14.0),
+    (12.0, 20.0),
+    (12.0, 24.0),
+    (12.0, 28.0),
+    (13.0, 26.0),
+    (13.0, 34.0),
+    (13.0, 34.0),
+    (13.0, 46.0),
+    (14.0, 26.0),
+    (14.0, 36.0),
+    (14.0, 60.0),
+    (14.0, 80.0),
+    (15.0, 20.0),
+    (15.0, 26.0),
+    (15.0, 54.0),
+    (16.0, 32.0),
+    (16.0, 40.0),
+    (17.0, 32.0),
+    (17.0, 40.0),
+    (17.0, 50.0),
+    (18.0, 42.0),
+    (18.0, 56.0),
+    (18.0, 76.0),
+    (18.0, 84.0),
+    (19.0, 36.0),
+    (19.0, 46.0),
+    (19.0, 68.0),
+    (20.0, 32.0),
+    (20.0, 48.0),
+    (20.0, 52.0),
+    (20.0, 56.0),
+    (20.0, 64.0),
+    (22.0, 66.0),
+    (23.0, 54.0),
+    (24.0, 70.0),
+    (24.0, 92.0),
+    (24.0, 93.0),
+    (24.0, 120.0),
+    (25.0, 85.0)
+  )
+
+  test("cars linear model") {
+    val m = for {
+      a <- Normal(0, 100).param
+      b <- Normal(0, 10).param
+      sigma <- Uniform(0, 30).param
+      _ <- Events.observe(cars) { s =>
+        val mu = a + b * s
+        Normal(mu, sigma)
+      }
+    } yield (a, b)
+
+    val w = m.waic(HMC(5), 100000, 10000, 1)
+    assert(w >= 400.0 && w <= 450.0)
```

Comment:
per Statistical Rethinking ch 6

Reply:
If you split WAIC into a separate object I think you'll be able to pass an array of samples that let you test an exact value.